edsl.md

```diff
@@ -153,9 +159,21 @@ where `F1 : F2 : F3 : F4` is the (two's complement) byte-array representation of
     rule #enc(   #bool( DATA )) => #padToWidth(32, #asByteStack(#getValue(   #bool( DATA ))))
 
     // dynamic Type
-    rule #enc(   #bytes(N, DATA)) => #enc(#uint256(N)) ++ #padToWidth(#ceil32(N), #asByteStack(DATA))
+    rule #enc(        #bytes(WS)) => #enc(#uint256(#sizeWordStack(WS))) ++ #padRightToWidth(#ceil32(#sizeWordStack(WS)), WS)
+    rule #enc(   #bytes(N, DATA)) => #enc(#uint256(N)) ++ #padRightToWidth(#ceil32(N), #asByteStack(DATA))
     rule #enc(#array(_, N, DATA)) => #enc(#uint256(N)) ++ #encodeArgs(DATA)
 
+    syntax WordStack ::= #padRightToWidth ( Int , WordStack ) [function]
+ // --------------------------------------------------------------------
+    rule  #padRightToWidth(N, WS) => #padRightToWidthAux(N -Int #sizeWordStack(WS), WS, .WordStack)
+
+    syntax WordStack ::= #padRightToWidthAux ( Int , WordStack , WordStack ) [function]
+ // -----------------------------------------------------------------------------------
+    rule #padRightToWidthAux(0, WS, ZEROS) => WS ++ ZEROS
+
+    rule #padRightToWidthAux(N, WS, ZEROS) => #padRightToWidthAux(N -Int 1, WS, 0 : ZEROS)
+      requires N >Int 0
+
```

Comment:
I think that `#padRightToWidth` should probably be put in `data.md` next to `#padToWidth` definition, @tomtomjhj.